code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -880,24 +880,39 @@ obj/item/assembly/radio_horn/receive_signal()
 //////////////////////////////////handmade shotgun shells//////////////////////////////////
 
 /obj/item/assembly/makeshiftshell
-
 	name = "filled pipe hulls"
 	desc = "Four open pipe shells, with propellant in them. You wonder what you could stuff into them."
 	icon_state = "Pipeshotrow"
+	var/thingsneeded = null //number of things still required to produce ammunition
+	var/crafttype = null //1 is glass, 2 is scrap, 3 is plasmaglass
 
 	attackby(obj/item/W, mob/user)
-		if(istype(W, /obj/item/raw_material/shard))
-			var/obj/item/ammo/bullets/pipeshot/glass/shot = new /obj/item/ammo/bullets/pipeshot/glass/(get_turf(src))
-			qdel(W)
-			qdel(src)
-			user.put_in_hand_or_drop(shot)
-
-		if(istype(W, /obj/item/raw_material/scrap_metal))
-			var/obj/item/ammo/bullets/pipeshot/scrap/shot = new /obj/item/ammo/bullets/pipeshot/scrap/(get_turf(src))
-			qdel(W)
-			qdel(src)
-			user.put_in_hand_or_drop(shot)
+		if (istype(W, /obj/item/raw_material/shard))
+			if (!crafttype)
+				crafttype = 1
+				thingsneeded = 2
+			if(crafttype == 1)
+				src.craftwith(W, src, user)
+		if (istype(W, /obj/item/raw_material/scrap_metal))
```

Comment:
One thing I've realized playing a couple of rounds trying to craft this. Scrap metal is extremely rare to find. The only way to obtain it is to purposely blow up flooring & walls. Or find it random-spawned in prefabs / maintenance flooring.

Reply:
That's intentional, to skew it's use towards antags and more dire situations/later in the round. On high pop I've usually been able to find 1-2 bits of scrap around the station/in debris even without blowing anything up, which is more or less how obtainable I want it to be.